src/main/java/com/travel/pay/facade/TravelTourFacade.java

```diff
@@ -0,0 +1,109 @@
+import one.util.streamex.StreamEx;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class TravelTourFacade {
+    private final PaymentService paymentService;
+    private final List<Tour> currentToursList;
+    private final List<Tour> intermediateToursList = new ArrayList<>();
+    private Tour currentTour;
+
+    public TravelTourFacade() {
+        this.paymentService = new PaymentServiceImpl();
+        TourService tourService = new TourServiceImpl();
+        this.currentToursList = tourService.getAllTours();
+    }
+
+    public void buyTour(String firstName, String lastName, String cardNumber, String expirationDate, String cvv) {
+        paymentService.payment(firstName, lastName, cardNumber, DateTimeUtils.formatCreditCard(expirationDate), cvv, currentTour.getPrice());
+        currentToursList.retainAll(intermediateToursList);
+
+        System.out.println(currentTour.withShortDescription());
+    }
+
+    public void getToursByDate(String tourDate) {
+
+        final List<Tour> tours = StreamEx.of(currentToursList)
+                .filter(tour -> tour.getFromDate().isEqual(DateTimeUtils.formatTourDate(tourDate)))
+                .toList();
+
+        StreamEx.of(tours)
+                .forEach(tour -> System.out.println(tour.withShortDescription()));
+
+        if (intermediateToursList != null) {
+            intermediateToursList.clear();
+            intermediateToursList.addAll(tours);
+        } else {
+            intermediateToursList.addAll(tours);
+        }
+    }
+
+    public void getToursByPrice(double price) {
+        final List<Tour> tours = StreamEx.of(intermediateToursList)
+                .filter(tour -> tour.getPrice() <= price)
+                .toList();
+
+        StreamEx.of(tours)
+                .forEach(tour -> System.out.println(tour.withShortDescription()));
+
+        if (intermediateToursList != null) {
+            intermediateToursList.clear();
+            intermediateToursList.addAll(tours);
+        } else {
+            intermediateToursList.addAll(tours);
+        }
+    }
+
+    public void selectTour(int id) {
+        final Optional<Tour> selectedTour = StreamEx.of(intermediateToursList)
+                .filter(tour -> tour.tourId() == id)
+                .findAny();
+
+        if (!selectedTour.isPresent()) {
+            System.out.println("Incorrect id number of tour");
+        } else {
+            selectedTour.ifPresent(System.out::println);
+            currentTour = selectedTour.get();
+        }
+    }
+
+    public void getCurrentRoomType() {
+        System.out.println(currentTour.getHotel().getAvailableRoomType());
+    }
+
+    public void selectRoomType(String typeRoom) {
+        Hotel.RoomType selectedRoomType = Hotel.RoomType.valueOf(typeRoom.toUpperCase());
+
+        final Optional<Hotel.RoomType> tR = StreamEx.of(Hotel.RoomType.values())
+                .filter(type -> type.equals(selectedRoomType))
```

Comment:
You could use findFirst and orElseThrow instead using if-else.

Reply:
At this step, I don't need to throw an exception for the user, because I can to provoke an early closed the app.